src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/TokenCertificateService.java

```diff
@@ -744,6 +746,35 @@ static boolean isCausedByCertNotFound(CodedException e) {
         return possibleActions;
     }
 
+    /**
+     * Deletes a collection of certificates
+     * @throws CertificateNotFoundException if certificate with given hash was not found
+     * @throws ActionNotPossibleException if delete was not possible due to cert/key/token states
+     */
+    public void deleteCertificates(List<CertificateInfo> certificateInfos)
+            throws CertificateNotFoundException, ActionNotPossibleException {
+        List<TokenInfo> tokenInfos = tokenService.getAllTokens();
+        for (CertificateInfo certificateInfo: certificateInfos) {
+            deleteCertificate(certificateInfo.getId(), tokenInfos);
+        }
+    }
```

Comment:
This is probably intentionally a non-atomic operation? Just wondering what happens in a case where some certs are deleted and some are not because in the middle of the deletion process a wild `CertificateNotFoundException` appears? 

In the case of removing orphans that shouldn't happen since the certs are not actually provided by the user. This is probably something to be looked more into if we ever need an API endpoint supporting the deletion of multiple certificates.

Reply:
Yeah, it is - kind of - intentional. If some of the certs have been deleted, and an error occurs midway, end result will be partly removed collection of certs. 

I believe the same thing would happen with the old UI implementation: https://github.com/nordic-institute/X-Road/blob/05e6443cc5212764fd68c537462a09c3750e57a9/src/proxy-ui/app/controllers/clients_controller.rb#L407-L426

I believe atomic deletion would be close to impossible since delete cert is a signer operation and core classes such as `TokenManager`, `SoftwareTokenWorker` and `HardwareTokenWorker` do not have a concepts similar to rollbackable transactions?